src/pages/Merchant/Register/Register.js

```diff
@@ -1,78 +1,80 @@
+import React, { useState } from "react";
 import { useNavigate } from "react-router-dom";
 
 import Button from "components/Button/Button";
 import InputControl from "components/InputControl/InputControl";
 
+import { validateEmail, validateMobile, validateOtp } from "utils/util";
+import { sendOtp, verifyOtp } from "utils/firebase";
+import { checkRegisterDetails, registerMerchant } from "api/user/register";
+
+import authLeftPanelImage from "assets/images/leftPanelImage.png";
+
 import styles from "./Register.module.scss";
 
 function Register() {
-  const leftPanelImage =
-    "https://image.shutterstock.com/z/stock-vector-set-of-fruit-and-vegetables-logo-for-groceries-agriculture-stores-packaging-and-advertising-318421853.jpg";
-
   const [errors, setErrors] = useState({});
+  const [otpPage, setOtpPage] = useState(false);
+  const [registerDetails, setRegisterDetails] = useState({});
+  const [otpObj, setOtpObj] = useState({});
+  const [values, setValues] = useState({
+    fname: "",
+    lname: "",
+    email: "",
+    mobile: "",
+    password: "",
+    confirmpass: "",
+    otp: "",
+  });
 
-  const fnameRef = useRef();
-  const lnameRef = useRef();
-  const emailRef = useRef();
-  const mobileRef = useRef();
-  const passRef = useRef();
-  const confirmpassRef = useRef();
   const navigate = useNavigate();
 
   const changeURl = () => {
-    navigate("/login");
+    navigate("/merchant/login");
   };
 
   const validateForm = () => {
     const dummyErrors = {};
-    if (fnameRef.current.value === "") {
+    if (values.fname === "") {
       dummyErrors.fname = "Enter first name";
     }
 
-    if (lnameRef.current.value === "") {
+    if (values.lname === "") {
       dummyErrors.lname = "Enter last name";
     }
 
-    if (mobileRef.current.value === "") {
+    if (values.mobile === "") {
       dummyErrors.mobile = "Enter mobile number";
-    } else {
-      if (
-        mobileRef.current.value.length !== 10 &&
-        !/^\d{10}$/.test(mobileRef.current.value)
-      ) {
-        dummyErrors.mobile = "enter valid mobile number";
-      }
+    } else if (!validateMobile(values.mobile)) {
+      dummyErrors.mobile = "Enter valid mobile number";
     }
 
-    if (emailRef.current.value === "") {
+    if (values.email === "") {
       dummyErrors.email = "Enter email";
     } else {
-      if (
-        !/^\w+([-]?\w+)*@\w+([-]?\w+)*(\w{2,3})+$/.test(emailRef.current.value)
-      ) {
+      if (!validateEmail(values.email)) {
         dummyErrors.email = "Enter valid email";
       }
     }
 
-    if (passRef.current.value === "") {
+    if (values.password === "") {
       dummyErrors.password = "Enter password";
     } else {
-      if (passRef.current.value.length < 8) {
+      if (values.password.length < 6) {
         dummyErrors.password = "Must be at least 8 characters";
       }
-      if (passRef.current.value.search(/[a-z]/i) < 0) {
+      if (values.password.search(/[a-z]/i) < 0) {
```

Comment:
add password validation in util so that we can use it in customer side also and then use that here also

Reply:
done